product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -184,15 +186,34 @@ private void validateIfProductWithSkuOrGtinOrSlugExist(String slug,
                                                            String gtin,
                                                            String sku) {
         if (isProductWithSlugAvailable(slug)) {
-            throw new DuplicatedException(Constants.ErrorCode.SLUG_ALREADY_EXISTED, slug);
+            throw new DuplicatedException(Constants.ErrorCode.SLUG_ALREADY_EXISTED_OR_DUPLICATED, slug);
         }
 
         if (isProductWithGtinAvailable(gtin)) {
-            throw new DuplicatedException(Constants.ErrorCode.GTIN_ALREADY_EXISTED, gtin);
+            throw new DuplicatedException(Constants.ErrorCode.GTIN_ALREADY_EXISTED_OR_DUPLICATED, gtin);
         }
 
         if (isProductWithSkuAvailable(sku)) {
-            throw new DuplicatedException(Constants.ErrorCode.SKU_ALREADY_EXISTED, sku);
+            throw new DuplicatedException(Constants.ErrorCode.SKU_ALREADY_EXISTED_OR_DUPLICATED, sku);
+        }
+    }
+
+    private void validateIfProductSkuOrGtinOrSlugDuplicated(List<ProductVariationPostVm> variations) {
```

Comment:
the method name should be "validateDuplicatedProduct" or something like that. We can reuse this method in the future.

Reply:
Thanks for the suggestion! I've just updated it to `validateProductVariationDuplicates`.